src/List.js

```diff
@@ -77,7 +78,14 @@ export function List() {
     const timeToMinDate = new Date() - minDate + ONE_MINUTE;
     //re-render the page so the item unchecks when it should be unchecked
     setTimeout(() => setReRender({}), timeToMinDate);
-  }, [reRender, items]);
+  }, [reRender, items, ONE_MINUTE]);
+
+  const handleDelete = async (id) => {
+    if (window.confirm('Are you sure you want to delete this item?')) {
+      await deleteDoc(doc(db, 'shopping-list', id));
+      setItems(items.filter((item) => item.id !== id));
```

Comment:
Here we have two different sources of truth for what are the items on the list...
   ...the items in the database (DB) 
   and 
   ...the items in local state.

This means that the values shown in our list don't necessarily reflect what's in the DB because we could imagine a situation in which multiple users are using the same list and performing different actions on it. In such case we'd run into a concurrency issue. This would also be the case with the `handleChange` call.

In order to avoid this, we could extract the data fetching logic into a function like it was previously and then call that function at the end of `handleDelete` and `handleChange`.

As with most things in programming there are trade-offs. One thing that makes a really good developer is the ability to identify trade-offs. This is something that interviewers look for in job interviews. In my experience, many less experienced devs make choices based on strongly held right/wrong opinions, while senior devs identify trade-offs and make choices based on what they think is best in a particular situation. 

So, as a fun exercise, what do you think are the trade-offs for the two approaches and which approach do you think that we should take in this situation with those trade-offs in mind?

Reply:
the trade off would be the cost of re-fetching items from the database.  i went ahead and changed the approach for the delete button.  I'm not sure I understand what would be needed for the handle change, I believe that the "checked" display is coming from the checked attribute on the component (!!item.lastPurchasedDate && new Date() - item.lastPurchasedDate < ONE_DAY) which i believe is based on the db call from the use effect.

Reply:
Ok, so I was looking into `handleChange` because I was kinda confused as to how it was working and it turns out that my initial assumption was totally off!
Apparently `onSnapshot` is WAY cooler than I thought. It is actually a reactive listening mechanism in which we're getting pushed updates rather than requesting and pulling them.
For that reason, it looks like we can actually completely scratch what I said.
We also don't need to call `setItems` at all in `handleDelete`. :-)

Reply:
got it ok, updated!